modules/deployment-model/src/main/java/org/mule/runtime/deployment/model/internal/plugin/BundlePluginDependenciesResolver.java

```diff
@@ -92,6 +102,29 @@ public BundlePluginDependenciesResolver(ArtifactDescriptorFactory<ArtifactPlugin
     return resolvedPlugins;
   }
 
+  private void verifyPluginExportedPackages(List<ArtifactPluginDescriptor> plugins) {
+    final Map<String, List<String>> exportedPackages = new HashMap<>();
+
+    boolean error = false;
+    for (ArtifactPluginDescriptor plugin : plugins) {
+      for (String packageName : plugin.getClassLoaderModel().getExportedPackages()) {
+        List<String> exportedOn = exportedPackages.get(packageName);
+
+        if (exportedOn == null) {
+          exportedOn = new LinkedList<>();
+          exportedPackages.put(packageName, exportedOn);
+        } else {
+          error = true;
+        }
+        exportedOn.add(plugin.getName());
+      }
+    }
+
+    if (error) {
+      throw new DuplicateExportedPackageException(exportedPackages);
```

Comment:
Is correct that some plugins export the same package which is exported on another plugin, but that won't break the build.
The thing is that the case you mentioned, HTTP exporting the same packages as Sockets, is already managed during the deployment on a "sanitization" process. Every package exported on a plugin A that is also exported on another B, when A depends on B, will be removed from the exported packages of A